### PR TITLE
No default OSGi private package to avoid publish build warnings

### PR DESCRIPTION
**Pull request by richardimaoka**
Refs #21899 

Disclaimer: I'm not very familiar with SBT and OSGi, so please let me know anything wrong here..

### what I found

The below warning:

```
[warn] bnd: Unused Private-Package instructions, no such package(s) on the class path: [com.typesafe.akka.distributed.data.experimental.*]
```

came from [bndtools/bnd -> Builder.java](https://github.com/bndtools/bnd/blob/1f3bafc331902bf16d47d083a36934b4960357b9/biz.aQute.bndlib/src/aQute/bnd/osgi/Builder.java#L569L571), which was used in  [sbtOsgi plugin's SbtOsgi.scala](https://github.com/sbt/sbt-osgi/blob/16ad2f8910530d24977efb9dc6cee40cbe619ae7/src/main/scala/com/typesafe/sbt/osgi/SbtOsgi.scala#L75L82).

`SbtOsgi` sets the default private package as :

```
  def defaultOsgiSettings: Seq[Setting[_]] = {
       ...
      bundleSymbolicName <<= (organization, normalizedName)(Osgi.defaultBundleSymbolicName),
       ...
      privatePackage <<= bundleSymbolicName(name => List(name + ".*")),
         //this becomes like "com.typesafe.akka.distributed.data.experimental.*"
         //or "com.typesafe.akka.testkit.*", etc
```

which is used in [akka's OSGi.scala]() as the default value for `OsgiKeys.privatePackage`, since it was not overriden.

This resulted in the unused (default) private packages.

### CODE REVIEW

**Comment by richard-imaoka** [comment]
The test failure was not relevant to this change?

```
[error] (akka-typed-experimental/pr-validation:test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 1808 s, completed Nov 30, 2016 7:41:03 PM
```

**Comment by johanandren** [comment]
Yeap, unrelated - `akka.typed.ActorContextSpec`

**Comment by richard-imaoka** [comment]
@johanandren Could I ask you to re run the tests?

**Comment by johanandren** [comment]
PLS BUILD

File: project/OSGi.scala [hunk 1]
```diff
@@ -18,7 +18,8 @@ object OSGi {
     packagedArtifact in (Compile, packageBin) <<= (artifact in (Compile, packageBin), OsgiKeys.bundle).identityMap,
     // This will fail the build instead of accidentally removing classes from the resulting artifact.
     // Each package contained in a project MUST be known to be private or exported, if it's undecided we MUST resolve this
-    OsgiKeys.failOnUndecidedPackage := true 
+    OsgiKeys.failOnUndecidedPackage := true,
+    OsgiKeys.privatePackage := Seq()
```
Comment (johanandren) [inline project/OSGi.scala]: Add a comment here about why. Something like "By default an entry is generated from module group-id, but our modules does not adhere to that such package naming"

**Review by johanandren — APPROVED** [review]
LGTM but a comment explaining would be good.

**Comment by richard-imaoka** [comment]
Sure, added a comment as suggested.

**Review by patriknw — APPROVED** [review]

**Pull request merged**